cpp/cmake_modules/SetupCxxFlags.cmake

```diff
@@ -373,6 +373,18 @@ if (NOT MSVC)
 
   set(C_FLAGS_RELEASE "-O3 -DNDEBUG")
   set(CXX_FLAGS_RELEASE "-O3 -DNDEBUG")
+
+  set(USE_RTTI_CHECKED_CAST ON)
+  if ("${COMPILER_FAMILY}" STREQUAL "gcc")
+    if ((COMPILER_VERSION VERSION_LESS "8.3") AND ((COMPILER_VERSION VERSION_GREATER "8.0") OR (OMPILER_VERSION VERSION_EQUAL "8.0")))
```

Comment:
`OR (OMPILER`

I think you need to references variables with `$` in here, not sure why cmake isn't complaining.

Reply:
No need for $ for the variables here. `VERSION_LESS` works with variable names. Although I hoped that it would be complaining about `OMPILER_VERSION`.